src/Text/Pandoc/Writers/ZimWiki.hs

```diff
@@ -44,20 +44,21 @@ import Data.Default (Default(..))
 import Network.URI ( isURI )
 import Control.Monad ( zipWithM )
 import Control.Monad.State ( modify, State, get, evalState )
---import Control.Monad.Reader ( ReaderT, runReaderT, ask, local )
 import Text.Pandoc.Class ( PandocMonad )
+import qualified Data.Map as Map
 
 data WriterState = WriterState {
     stItemNum   :: Int,
-    stIndent    :: String          -- Indent after the marker at the beginning of list items
+    stIndent    :: String,         -- Indent after the marker at the beginning of list items
+    stInTable   :: Bool            -- Inside a table
   }
 
 instance Default WriterState where
-  def = WriterState { stItemNum = 1, stIndent = "" }
+  def = WriterState { stItemNum = 1, stIndent = "", stInTable = False }
 
 -- | Convert Pandoc to ZimWiki.
 writeZimWiki :: PandocMonad m => WriterOptions -> Pandoc -> m String
-writeZimWiki opts document = return $ evalState (pandocToZimWiki opts document) (WriterState 1 "")
+writeZimWiki opts document = return $ evalState (pandocToZimWiki opts document) (WriterState 1 "" False)
```

Comment:
I think what @tarleb is proposing is simply: `evalState (pandocToZimWiki opts document) def`